cmake/msvc-toolchain.cmake

```diff
@@ -0,0 +1,19 @@
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+include_guard(GLOBAL)
+
+set(CMAKE_C_COMPILER cl)
+set(CMAKE_CXX_COMPILER cl)
+
+if(BEMAN_BUILDSYS_SANITIZER STREQUAL "ASan")
+    set(CMAKE_CXX_FLAGS_DEBUG_INIT "/fsanitize=address /Zi")
+    set(CMAKE_C_FLAGS_DEBUG_INIT "/fsanitize=address /Zi")
```

Comment:
Is `/Zi` is related to ASan? Should it be enabled in all build modes?

Reply:
Yes. 

Do I need to include documentation for this? 

Omitting this flag will cause warning C5072.

MSVC side documentation:
https://learn.microsoft.com/en-us/cpp/error-messages/compiler-warnings/compiler-warning-c5072?view=msvc-170

Reply:
If they fix it, we could remove the flag? "Why" comments are useful.